### ISSUE TITLE
client/asset/btc/electrum .. Firo-Electrum `payto` command is different

### ISSUE THREAD

**Issue by dev-warrior777** [decred/dcrdex#2359]
Firo-Electrum is based on https://github.com/dashpay/electrum-dash-old/ which is old. Although Firo-Electrum is 2487 commits ahead the `payto` method does not contain an `rbf` parameter unlike latest Electrum for bitcoin.

```
bitcoin:
payto(self, destination, amount, fee=None, feerate=None, from_addr=None, from_coins=None, change_addr=None, nocheck=False, unsigned=False, 
**rbf=None,** password=None, locktime=None, addtransaction=False, wallet: Abstract_Wallet = None):

firo:
payto(self, destination, amount, fee=None, feerate=None, from_addr=None,from_coins=None, change_addr=None, nocheck=False, unsigned=False, 
password=None, locktime=None, addtransaction=False, wallet: Abstract_Wallet = None):

```
Sending a request to this API from dex electrum code uses the struct below

![image](https://github.com/decred/dcrdex/assets/126670673/bd14b01a-4055-427d-887e-de525cb52ad1)

I have commented out the original RBF parameter and introduced one possible fix by adding `,omitempty`.

- Testing with firo electrum works ok
- Testing with bitcoin electrum is not entirely satisfactory although it works

My preferred solution is to make an alternative version of the `paytoReq` struct above maybe called `paytoOldReq` with no RBF field at all.

I tried hacking the other end in python code by adding a dummy `rbf` parameter and that works but it could be a breaking change for current users so I rejected that after discussion with Firo core devs

**_Please advise_**

**Comment by dev-warrior777** [comment]
correct

**Comment by dev-warrior777** [comment]
Doge electrum - probably unused - last changes 9 years ago
Dash electrum - pretty old - last changes 6 years ago

So a new base struct would be the intersection of electrum for Bitcoin/Litecoin/BCH/Firo parameters for `payto` most of which are optional.

The key here is that sending a json param that does not exist in python errors. Not sending an optional parameter should always be okay.

**Comment by ukane-philemon** [comment]
> The key here is that sending a json param that does not exist in python errors. 

With the `omitempty` tag, I believe this won't be an issue, since Go will not add the field when marshalling to json. Electrum wallets should simply not provide a value for fields they don't care about (assuming they all share the same payto struct ¯\_(ツ)_/¯).

**Comment by chappjc** [comment]
> Updating the UI is another sticky option

Let's keep RBF off the UI entirely.  All sends would ideally use false.

**Comment by dev-warrior777** [comment]
![image](https://github.com/decred/dcrdex/assets/126670673/e5dd3a25-dd28-41ee-92ed-d3c5a5040c80)
Going here for now .. unfinished

**Comment by chappjc** [comment]
> Going here for now .. unfinished

OK.  You probably don't need to name `BaseReq` though.  Just embed it so it marshals flat.

**Comment by dev-warrior777** [comment]
Ok thank you for your better suggestion.

**Comment by dev-warrior777** [comment]
> > Updating the UI is another sticky option
> 
> Let's keep RBF off the UI entirely. All sends would ideally use false.

https://github.com/spesmilo/electrum/issues/8088

**Comment by dev-warrior777** [comment]
![image](https://github.com/decred/dcrdex/assets/126670673/be902055-6397-4237-af93-8ded7557b000)

Bitcoin specific:
Make sure you set `use_rbf` false on the `config` from the electrum console Or set it in the config file before starting. A change in the electrum console updates the config file, even if the change is from say False -> False
This changes the behavior for RBF

However...
![image](https://github.com/decred/dcrdex/assets/126670673/1b5ed542-76db-48a4-9cea-fe09758e8d64)
Checking/unchecking this before sending is not honored. The config value is used which can be checked by dumping the raw transaction and looking at the nSequence value.


So if you only care about electrum bitcoin then this is fine: 
![image](https://github.com/decred/dcrdex/assets/126670673/bb4349e0-5422-42e6-bd52-aae5a2f9af63)
Just make sure the config value `use_rbf` is what you want

Now I will re-check Firo & BCH again (now I have repaired my dot bitcoin directories wrecked by bitcoincash)

**Comment by dev-warrior777** [comment]
**BCH:**
`payto` 
- No wallet path input
- No RBF input

_Sends back more complex json than just the Tx string_

![image](https://github.com/decred/dcrdex/assets/126670673/1b4042c7-ea25-4b3b-a2d4-5db222fefe56)

More useful for future is that the `broadcast`  python method is also more complex (see above)

**Comment by dev-warrior777** [comment]
**RBF: Conclusion:**
This parameter can be updated to _json_ `omitempty` or disappear altogether, unless useful for Dash. RBF defaults to config setting for electrum bitcoin & does not exist for Firo or BCH.
It would be useful if someone can update RBF in git master

**Other:** 
There are differences in how the `feerate` mechanisms work for each of the tested electrum wallets. Nothing that cannot be handled I think.

Anyway that is another story .. I am closing this Issue.
